src/dut.rs

```diff
@@ -307,13 +308,23 @@ impl DutInfo {
             }
         }
         if keys.contains(&"dut_id") {
-            let model = values.get("model");
-            let serial = values.get("serial");
-            if let (Some(Ok(model)), Some(Ok(serial))) = (model, serial) {
+            let serial: String = if let Some(Ok(serial)) = values.get("serial") {
+                serial.to_string()
+            } else {
+                // Some DUTs don't have serial number. So use MAC address,
+                let serial = if let Some(Ok(mac)) = values.get("mac") {
+                    mac.replace(':', "")
+                } else {
+                    return Err(anyhow!("Failed to get MACaddr. {:?}", values.get("mac")));
+                };
+                values.insert("serial".to_string(), Ok(serial.to_string()));
```

Comment:
the "pseudo serial" should not be stored as serial here, as it is not a serial. (we should use dut_id everywhere to identify a DUT)

Reply:
But without serial, the command doesn't work. See https://github.com/google/lium/blob/4f24e79a498d17a4de9ed762700eafa2f17dcec2/src/dut.rs#L326

Reply:
That is because serial will be the dependency of dut_id (which is not true anymore with this change)

https://github.com/google/lium/blob/ca94eaad10db333982e2afe94bd73e64b84fa1b4/src/dut.rs#L345

so you can remove the line above to make it work ;)

Reply:
If I removed only that requirement, `lium dut list --add` failed. If `serial` is not found.
```
Error: failed to get key serial: Some(Err(key serial found but was empty. stderr: [WARN] VPD partition not formatted. It's fine.
findString(): Vpd data 'serial_number' was not found.))
```
As I pointed, the `parse_values()` checks the `DEFAULT_DUT_INFO_KEYS` at the last loop, and if it returns an error, it shows the error.

Reply:
https://github.com/google/lium/blob/ca94eaad10db333982e2afe94bd73e64b84fa1b4/src/dut.rs#L181-L190

Ah, then please remove `serial` from `DEFAULT_DUT_INFO_KEYS` as well, since the assumption (`serial` is one of the "keys that are always available") was wrong.

Reply:
No, it doesn't work.
```
$ lium dut info --dut geralt_NoSeriala0cec8e67395 
Fetching info for SshInfo { host: "2401:fa00:480:ee08:a2ce:c8ff:fee6:7395", port: 22 }...
Error: failed to get key serial: Some(Err(key serial found but was empty. stderr: [WARN] VPD partition not formatted. It's fine.
findString(): Vpd data 'serial_number' was not found.))
```
With following change.
```
--- a/src/dut.rs
+++ b/src/dut.rs
@@ -179,13 +179,12 @@ const CMD_GET_DEFAULT_IFACE: &str =
     r"ip route get 8.8.8.8 | sed -E 's/^.* dev ([^ ]+) .*$/\1/' | head -n 1";
 
 // Only keys that are always available can be listed here
-const DEFAULT_DUT_INFO_KEYS: [&str; 8] = [
+const DEFAULT_DUT_INFO_KEYS: [&str; 7] = [
     "timestamp",
     "dut_id",
     "hwid",
     "release",
     "model",
-    "serial",
     "board",
     "mac",
 ];
@@ -320,7 +319,6 @@ impl DutInfo {
                         values.get("mac")
                     ));
                 };
-                values.insert("serial".to_string(), Ok(serial.to_string()));
                 serial
             };
             if let Some(Ok(model)) = values.get("model") {
```

Reply:
ok, let 's do so...